01-ruby-fundamentals/keyword-arguments.md

```diff
@@ -157,39 +205,38 @@ coffee_price(:drip, :tall, cold: true)
 coffee_price(:cappuccino, :grande)
 ```
 
-The only things we've changed are the parameters in the method signature, the body of the method remains the same. The end result should resemble [this gist](https://gist.github.com/droberts-ada/07714ad0e2a3bc5a60c50d6a78e02f15).
-
-There are a few things to note here:
-- In the method signature, each keyword argument comes with a _default value_
-  - The syntax is similar to keys and values in a hash
-- When you invoke a method, keyword arguments are optional
-  - If you omit one, the value from the method signature will be used instead
-- When a method is invoked, any keyword arguments are passed by name (a.k.a. _keyword_)
-- Keyword arguments can be passed in any order.
-- Keyword arguments can co-exist with regular arguments (also called _positional arguments_)
-  - Positional arguments must come first, both in the definition and when the method is called
-  - All positional arguments must be supplied for every call
+In each of these iterations, the only things we've changed are the parameters in the method signature, the body of the method remains the same. The end result should resemble [this gist](https://gist.github.com/droberts-ada/07714ad0e2a3bc5a60c50d6a78e02f15).
 
-### When to use Keyword Arguments
+### When to use which argument types
 
-Keyword arguments aren't always the right choice.
+**Optional arguments** are handy but aren't always the right choice.
 
 - Sometimes an argument doesn't have a clear default value
   - For `type`, should the default be `:drip`, `:latte` or `:cappuccino`?
 - Sometimes a method doesn't make sense without one of the arguments
   - How can you price coffee if you don't know the size?
 
-In these cases, positional arguments are usually the right choice. Keyword arguments are useful when a method has an alternate way of behaving, or a long list of optional arguments.
+**Keyword arguments** are mostly useful when the number of arguments becomes more than 2 or 3.
 
-The difference between the two is similar to the difference between options (flags) and arguments to a command-line program. Options change _how_ the program behaves, arguments change _what_ it acts on.
+When you make a method, and you write code to call it, ask yourself, "is it hard remember or understand which argument is which?" or conversely "is it more trouble than it's worth to type out the keywords?"
 
 ## Summary
 
+|   | Required | Optional |
+|---|----------|----------|
+| **Positional** | coffee_price(type, size, extra_shots, cold) | coffee_price(type, size, extra_shots = 0, cold = false) |
+| **Keyword** | coffee_price(type:, size:, extra_shots:, cold:) | coffee_price(type, size, extra_shots: 0, cold: false) |
```

Comment:
Nit: Can we line these up with spaces and `-`s so that we can make it look more like an ascii-art table?

Reply:
I want to clarify that it looks pretty in an actual markdown editor, so I didn't think of it as worth the effort, but I certainly can

Reply:
<img width="951" alt="Screen Shot 2019-07-29 at 11 12 00 AM" src="https://user-images.githubusercontent.com/52671000/62071610-ca40f480-b1f1-11e9-9d47-d3af554915a6.png">

Reply:
@kaidamasaki Is there a team tenet around how markdown tables are supposed to look? Should we drive toward forming one?